includes/express-checkout/class-wc-payments-express-checkout-button-helper.php

```diff
@@ -760,22 +760,33 @@ private function is_product_supported() {
 		 *
 		 * @psalm-suppress UndefinedClass
 		 */
-		if ( is_null( $product )
-			|| ! is_object( $product )
-			|| ! in_array( $product->get_type(), $this->supported_product_types(), true )
-			|| ( class_exists( 'WC_Subscriptions_Product' ) && $product->needs_shipping() && WC_Subscriptions_Product::get_trial_length( $product ) > 0 ) // Trial subscriptions with shipping are not supported.
+
+		if ( is_null( $product ) || ! is_object( $product ) ) {
+			$is_supported = false;
+		} else {
+			// Simple subscription that needs shipping with free trials is not supported.
+			$is_free_trial_simple_subs = class_exists( 'WC_Subscriptions_Product' ) && $product->get_type() === 'subscription' && $product->needs_shipping() && WC_Subscriptions_Product::get_trial_length( $product ) > 0;
+
+			// Disable ECE for all variable subscriptions with free trials, as they are not currently supported. For now, ECE will be disabled for all such cases, and a solution will be addressed in a separate PR.
+			$is_free_trial_variable_sub = class_exists( 'WC_Subscriptions_Product' ) && $product->get_type() === 'variable-subscription' && WC_Subscriptions_Product::get_trial_length( $product ) > 0;
```

Comment:
**If a product has two variations and only one includes a free trial, how should the ECE buttons behave?** In my tests, the buttons aren’t displayed for any variation if at least one has a trial period.

Reply:
It should work as it did previously. The issue you mentioned will be addressed in the next PR.

Reply:
^ This line will actually be either modified or removed in my next PR since I will enable ECE for those kind of subscriptions.

Reply:
Ok, I confirm it still works the same as before.

I think the variables here can be simplified since they’re repeating themselves. In this PR, we’re disabling ECE for all subscriptions with free trials, so we only need:

```php
$is_subscription_with_free_trial = class_exists( 'WC_Subscriptions_Product' ) 
    && WC_Subscriptions_Product::is_subscription( $product ) 
    && WC_Subscriptions_Product::get_trial_length( $product ) > 0;
```
In the context of this PR, the distinction of subscription type doesn’t seem relevant, so we can completely ignore it.

Reply:
Nevermind, I just realized that for simple subscriptions, there’s the additional shipping factor to consider.